mcu-util/src/orb/main_board.rs

```diff
@@ -66,15 +66,26 @@ impl MainBoardBuilder {
         // Send a heartbeat to the main mcu to ensure it is alive
         // & "subscribe" to the main mcu messages: messages to the Jetson
         // are going to be sent after the heartbeat
-        canfd_iface
+        let _ = canfd_iface
             .send(McuPayload::ToMain(
                 main_messaging::jetson_to_mcu::Payload::Heartbeat(
                     main_messaging::Heartbeat {
                         timeout_seconds: 0_u32,
                     },
                 ),
             ))
-            .await?;
+            .await
+            .map(|c| {
+                if let CommonAckError::Success = c {
+                    Ok(())
+                } else {
+                    Err(eyre!("ack error: {c}"))
+                }
+            })
+            .unwrap_or_else(|e| {
+                error!("Failed to send heartbeat to main mcu: {:#?}", e);
+                Err(eyre!("Failed to send heartbeat to main mcu"))
+            });
```

Comment:
You're producing an error just to immediately discard it.

I would rework this code to be something like:

```rust
        let ack_result = canfd_iface
            .send(McuPayload::ToMain(
                main_messaging::jetson_to_mcu::Payload::Heartbeat(
                    main_messaging::Heartbeat {
                        timeout_seconds: 0_u32,
                    },
                ),
            ))
            .await
            .map(|c| {
                if let CommonAckError::Success = c {
                    Ok(())
                } else {
                    Err(eyre!("ack error: {c}"))
                }
            });
      if let Err(e) = ack_result
           error!("Failed to send heartbeat to main mcu: {:#?}", e);
      });
```

Reply:
I pushed a commit for this just now.